src/Uno.UI/UI/Xaml/UIElement.Pointers.wasm.cs

```diff
@@ -241,6 +241,21 @@ public static void OnNativePointerEvent()
 		}
 	}
 
+	private static readonly HashSet<uint> s_activePointers = new();
+	private static Dictionary<uint, uint> s_nativeToManagedPointerId = new();
+
+	private static uint TransformPointerId(uint nativePointerId)
+	{
+		if (s_nativeToManagedPointerId.TryGetValue(nativePointerId, out var pointerId))
+		{
+			return pointerId;
+		}
+
+		pointerId = s_activePointers.Count == 0 ? 1 : s_activePointers.Max() + 1;
```

Comment:
Also: I think that the `Max()` will enumerate the collection. It would be great to just reset to 0 when we remove the last ID from the `s_nativeToManagedPointerId`.

Reply:
You can keep a `_activePointerCount` that you reset to 0 only when you clear the `_activePointers` list :)

Reply:
I think that's not equivalent, unless I'm missing something.

Consider two cases:

#### Case 1

- press finger 1
- press finger 2
- release finger 1
- press finger 3 (here I expect the id to be 3)

#### Case 2

- press finger 1
- press finger 2
- release finger 2
- press finger 3 (here I expect the id to be 2)

If I used `_activePointerCount`, the id will be 3 in both cases I think?

Reply:
Yes the ID should be 3, but here it depends if you decrement the `_activePointerCount` when finger 2 is released or not. You should not and reset it to 0 only when all fingers are removed from the screen (i.e. `s_activePointers is {Count: 0}`).

I agree that `_activePointerCount` might not be the right name though.

Reply:
@dr1rrb Fixed in last commit. Not sure if the logic I wrote is correct or if it matches your intent. Can you take a look? Thanks!